app/queues/dns/create-record-worker.server.ts

```diff
@@ -0,0 +1,142 @@
+import { RecordStatus } from '@prisma/client';
+
+import type { DomainRecord } from '~/lib/domains.server';
+
+export const addDbRecordQueueName = 'add-db-record';
+export const createDnsRecordQueueName = 'create-dns-record';
+export const checkDnsStatusQueueName = 'check-dns-status';
+export const syncDbStatusQueueName = 'sync-db-status';
+
+export const addDbRecordWorker = new Worker<
+  Pick<DomainRecord, 'username' | 'type' | 'name' | 'value'>
+>(
+  addDbRecordQueueName,
+  async (job) => {
+    logger.info('Creating a record in DB');
+    const { username, type, name, value } = job.data;
+
+    if (!username) {
+      throw new Error('Username is not provided');
+    }
+
+    const existingRecords = await prisma.record.count({
+      where: {
+        type,
+        name,
+        value,
+      },
+    });
+
+    if (existingRecords > 0) {
+      throw new Error('Record already exists');
+    }
+
+    try {
+      const result = await prisma.record.create({
+        data: {
+          username,
+          type,
+          name,
+          value,
+          expiresAt: setMonthsFromNow(6),
+          status: RecordStatus.pending,
+        },
+      });
+      return result.id;
+    } catch (error) {
+      logger.warn('Could not create a record in DB', { error });
+    }
+  },
+  { connection: redis }
+);
+
+export const createDnsRecordWorker = new Worker<
+  Pick<DomainRecord, 'username' | 'type' | 'name' | 'value'>
+>(
+  createDnsRecordQueueName,
+  async (job) => {
+    logger.info('Creating a record in Route 53');
+    const { username, type, name, value } = job.data;
+
+    try {
+      return await createRecord(username, type, name.toLowerCase(), value);
+    } catch (error) {
+      logger.warn('Could not create a record in Route53', { error });
+    }
+  },
+  { connection: redis }
+);
+
+export const checkDnsStatusWorker = new Worker(
+  checkDnsStatusQueueName,
+  async (job) => {
+    logger.info('Checking Route53 readiness');
+
+    const values: any = await job.getChildrenValues();
+    const key = Object.keys(values)[0];
+    const changeId = values[key];
+    let status: string;
+
+    do {
+      status = await getChangeStatus(changeId);
+      await delay(1000);
+    } while (status === 'PENDING');
+    return status;
+  },
+  { connection: redis }
+);
+
+export const syncDbStatusWorker = new Worker(
+  syncDbStatusQueueName,
+  async (job) => {
+    logger.info('Sync DB status with DNS');
+    const values = await job.getChildrenValues();
+    const keys = Object.keys(values);
+
+    let recordId, status;
+    if (keys.length !== 2) {
+      throw new Error('Invalid children values from previous workers');
+    } else {
+      if (keys[0].includes(addDbRecordQueueName)) {
+        recordId = values[keys[0]];
+        status = values[keys[1]];
+      } else {
+        recordId = values[keys[1]];
+        status = values[keys[0]];
+      }
+    }
+
+    try {
+      switch (status) {
```

Comment:
Simplify this, and move the condition into the value for the `status` vs. a switch:

```ts
await prisma.record.update({
  where: {
    id: recordId,
  },
  data: {
    status: status === 'INSYNC' ? RecordStatus.active : RecordStatus.error,
  },
});
```

Reply:
Yes, it's definitely cleaner. Thank you